openquakeplatform/openquakeplatform/models.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2015, GEM Foundation.
+#
+# This program is free software: you can redistribute it and/or modify
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from django.contrib.gis.db import models
+from django.db.models.sql.constants import QUERY_TERMS
+from django.db.backends.postgresql_psycopg2.base import DatabaseWrapper
+from django.db.backends.postgresql_psycopg2.operations import (
+    DatabaseOperations)
+
+QUERY_TERMS.add('unaccent')
+DatabaseWrapper.operators['unaccent'] = '=~@ %s'
+
+
+def my_lookup_cast(self, lookup_type):
+    """
+    Adding 'unaccent' to the standard lookup_cast
+    """
+    lookup = '%s'
+
+    # Cast text lookups to text to allow things like filter(x__contains=4)
+    if lookup_type in ('iexact', 'contains', 'icontains', 'unaccent',
+                       'startswith', 'istartswith', 'endswith', 'iendswith'):
+        lookup = "%s::text"
+
+    # Use UPPER(x) for case-insensitive lookups; it's faster.
+    if lookup_type in ('iexact', 'icontains', 'unaccent',
+                       'istartswith', 'iendswith'):
+        lookup = 'UPPER(%s)' % lookup
+
+    return lookup
+
```

Comment:
But fortunately in next releases Django natively provides the accent-insensitive filtering, so we will remove our customization and use the standard way.